.github/dependabot.yml

```
@@ -0,0 +1,34 @@
+# Individual dependency updates with security focus
+version: 2
+updates:
+  # Security updates - daily
+  - package-ecosystem: "npm"
+    directory: "/"
+    schedule:
+      interval: "daily"
+    open-pull-requests-limit: 5
+    labels:
+      - "security"
+    groups:
+      security-patches:
+        applies-to: security-updates
+        patterns:
+          - "*"
+
+  # All dependencies - weekly, individual PRs
+  - package-ecosystem: "npm"
+    directory: "/"
+    schedule:
+      interval: "weekly"
+    open-pull-requests-limit: 10
```

> increase-if-necessary: Only update if required to satisfy constraints
Only bumps versions when needed by your dependency constraints. For example, if your package.json has "lodash": "^4.17.0" and both 4.17.21 and 5.0.0 are available, it will only update to 4.17.21 (not the major    version 5.0.0) since your semver range is satisfied.
  
```suggestion
    versioning-strategy: "increase-if-necessary"
    open-pull-requests-limit: 10
```